pkgs/development/python-modules/mmcv/default.nix

```diff
@@ -0,0 +1,92 @@
+{ lib
+, buildPythonPackage
+, fetchFromGitHub
+, pytest
+, pythonOlder
+, cudaSupport ? false
+, cudatoolkit
+, torchCudaArchList ? "Kepler"
+, torch
+, torchvision
+, torch-bin
+, torchvision-bin
+, opencv4
+, yapf
+, packaging
+, pillow
+, addict
+, ninja
+, which
+, onnx
+, onnxruntime
+, scipy
+, pyturbojpeg
+, tifffile
+, lmdb
+}:
+
+buildPythonPackage rec {
+  pname = "mmcv";
+  version = "1.7.0";
+  format = "setuptools";
+
+  disabled = pythonOlder "3.6";
+
+  src = fetchFromGitHub {
+    owner = "open-mmlab";
+    repo = pname;
+    rev = "refs/tags/v${version}";
+    hash = "sha256-EVu6D6rTeebTKFCMNIbgQpvBS52TKk3vy2ReReJ9VQE=";
+  };
+
+  torch-deps = if cudaSupport then [ torch-bin torchvision-bin ] else [ torch torchvision ];
+  nativeBuildInputs = [ ninja which ]
+    ++ lib.lists.optionals cudaSupport [ cudatoolkit.lib cudatoolkit.out cudatoolkit.cc ] ++ torch-deps;
+
+  postPatch = ''
+    substituteInPlace setup.py --replace "cpu_use = 4" "cpu_use = $NIX_BUILD_CORES"
+  '';
+
+  # reason for not using pytestCheckHook is similiar to what
+  # is already mentioned in pkgs/development/python-modules/typed-ast
+  # test_cnn test_ops really requires gpus to be useful.
+  # some of the tests take exceedingly long time.
+  # the rest of the tests are disabled due to sandbox env.
```

Comment:
This comment does not justify not using pytestCheckHook at all

```suggestion
```

Reply:
I try to do ```
  preCheck = ''
    PYTHONPATH=$out/${python.sitePackages}:$PYTHONPATH
  ''

```
which seems to work with a lot of packages with c extension, but it didn't work for me.

Reply:
Another common way is to rm the the conflicting source directory but pytestCheckHook does just invoke pytest as you do, it only builds the arguments to it.

Reply:
had to find that out the hard way. lol

Reply:
done